utils/ibdmp-decode

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/python3
+
+import base64
+import collections
+import hashlib
+import re
+import sys
+
+DEBUG = False
+RE_HEADER = r'_ibdmp:[0-9]+/[0-9]+[|]chunks=[0-9]+,md5=[0-9a-f]+[|]'
+RE_CHUNK = r'_ibdmp:[0-9]+/[0-9]+[|][0-9]+:[A-Za-z0-9+/=]+[|]'
+
+
+def USAGE():
+    self = sys.argv[0].split('/')[-1]
+    lines = [
+        "usage: %s path/to/console.log path/to/target.tar.xz" % self,
+        "",
+        "Decode debug tarball emited by leapp's initramfs in-band",
+        "console debugger, ibdmp().",
+    ]
+    sys.stderr.writelines('%s\n' % l for l in lines)
+    sys.exit(2)
+
+
+def LOG_DEBUG(msg):
+    if DEBUG:
+        sys.stderr.write('DEBUG: %s\n' % msg)
+
+
+def LOG_WARN(msg):
+    sys.stderr.write('%s\n' % msg)
+
+
+class IbdmbDecodeError(ValueError):
+    pass
+
+
+class UsageError(ValueError):
+    pass
+
+
+class _Chunk:
+
+    @classmethod
+    def from_raw1(cls, raw_chunk):
+        assert raw_chunk.startswith('_ibdmp:')
+        areas = raw_chunk.split('|')
+        parts = areas[1].split(':')
+        return cls(
+            ordinal=int(parts[0]),
+            payload=str(parts[1]),
+        )
+
+    def __init__(self, ordinal, payload):
+        self.ordinal = ordinal
+        self.payload = payload
+
+    def __hash__(self):
+        return hash((self.ordinal, self.payload))
+
+
+class Header:
+    """
+    Chunk set header
+    """
+
+    @classmethod
+    def from_rawN(cls, raw_headers):
+        """
+        Initialize chunk header from header chunk candidates
+
+        raw_headers is a list of strings that contain encoded chunk
+        parameters for the whole chunk set, ie. number of chunks, number
+        of iterations, and MD5 hash of the content encoded in the chunk set.
+
+        Raw header chunks can be corrupted so this factory will choose
+        winner based on prevalence.
+
+        For chunk set example in ChunkCounter.__init__ corresponding
+        raw headers could look similar to this:
+
+            _ibdmp:1/3|chunks=2,md5=281cc34e13cb4a502abd340fd07c4020|
+            _ibdmp:2/3|chunks=2,md5=281cc34e13cb4a502abd340fd07c4020|
+            _ibdmp:3/3|chun?s=2,md5=281cc34e13cb4a502abd340fd07c4020|
+
+        In this case, the winner is the first and second one.
+        """
+        cntr = collections.Counter([
+            Header._from_raw1(rh) for rh in raw_headers
+        ])
+        if not cntr:
+            LOG_WARN("no dumps found in this console log")
+            raise IbdmbDecodeError()
+        winner = cntr.most_common()[0][0]
+        LOG_DEBUG("header winner: %s" % winner)
+        return winner
+
+    @classmethod
+    def _from_raw1(cls, raw_header):
+        parts = raw_header.split('|')
+        _, stats = parts[0].split(':')
+        pairs = parts[1].split(',')
+        try:
+            assert pairs[0].startswith('chunks=')
```

Comment:
Don't `assert` → `except AssertionError` in non-test code. Assertions are skipped when Python is running with `-O` or `-OO` so they shouldn't be relied on for control flow (pretty much as all exception handling).

Reply:
OK, 18c9380 replaces asserts (here but also in `_Chunk.from_raw1` around L50) with `if` and uses "classic" `LOG_WARN` + `IbdmbDecodeError` combo.

And yes, it's `IbdmbDecodeError`, and yes, it's `b`, and yes, it's the same f-ing typo again -- see, @pirat89 ? -- even pylint does not help :D  fixed in 430badd